pkg/ui/buildutil/hash_files.go

```diff
@@ -0,0 +1,55 @@
+// Copyright 2022 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package buildutil
+
+import (
+	"crypto/sha1"
+	"encoding/hex"
+	"io"
+	"io/fs"
+
+	"github.com/cockroachdb/errors"
+)
+
+// HashFilesInDir recursively computes the SHA1 hash of every file in fsys
+// starting at root, and stores the computed hashes in dest["/path/to/file"]
+// (*including* a leading "/").
+func HashFilesInDir(dest *map[string]string, fsys fs.FS) error {
+	if dest == nil {
+		return errors.New("Unable to hash files without a hash destination")
+	}
```

Comment:
For some reason, github shows me a few files in this diff as being indented by one or two characters, but this definitely uses tab characters when I `cat` them locally (or when viewed through reviewable).  I've no idea what's causing that.

![image](https://user-images.githubusercontent.com/665775/160027677-f1da4984-7aef-4413-9472-548e82f8de4e.png)